### DISCUSSION TITLE
Auto Dark Mode

### DISCUSSION THREAD

**Discussion by Olaf-007** [godotengine/godot-proposals#4924] [Editor]
As I am working on my game projects in the afternoon, when the sun sets, my Computer turns into Darkmode to go easy on eyes. However, whatever project I am working on, I always have to manually switch into dark mode and then, the next day, I have to manually switch back to light mode.
Now: I would love to have an option to enable auto dark mode so that when the registry key changes, the engine recognizes it and switches too.
It would look like this: 
![auto dark mode option](https://user-images.githubusercontent.com/56698992/179942689-e8576c36-5488-4273-a75a-4f64f2a8d659.png)
So you have three options: one to enable the auto dark mode, one to chose a light theme and one for the dark theme. 
What do you think?
This issue is certainly related to the problem that the Windows title bar does not change appearance when in Windows dark mode: https://github.com/godotengine/godot-proposals/issues/1868

**Comment by Calinou** [comment]
There are two problems with automatically using a dark theme in an application that uses a dark theme by default:

- All operating systems default to a light theme. This means that Godot would also have to default to a light theme, which is a no-go.
- Detecting the OS theme requires complex OS-specific code. On Windows, the relevant APIs are only exposed on recent Windows 10 versions and Windows 11. Linux probably doesn't have a standard API for this.

Instead, what we could do is add a keyboard shortcut to toggle between a light and dark color scheme. This requires https://github.com/godotengine/godot-proposals/issues/2938 to be implemented to work with custom theme presets though.

**Reply by Olaf-007** [reply]
To 1) The auto dark mode option does not has to be enabled by default. If it is enabled by the user, Godot would automatically switch to the theme light or dark depending on the time of the day (or rather of the environment setting of the OS). Would this conquer the no-go? 
2) Ok! I see why this was not done yet. I appreciate a lot that the Godot Editor is tremendously lightweight because it does not implement specific code. However, if one wants to export a project to Android the first time, Godot additionally downloads an Android package. May this be a solution for a OS-specific download? 
3) I'd be quite fine with a shortcut already. It is a great suggestion! This, though, would still need the two dark and light theme settings as you requested it in the named issue.

**Reply by Calinou** [reply]
> To 1) The auto dark mode option does not has to be enabled by default. If it is enabled by the user, Godot would automatically switch to the theme light or dark depending on the time of the day (or rather of the environment setting of the OS). Would this conquer the no-go?

Like most creative applications (Blender, Adobe, …), a lot of people are used to Godot defaulting to a dark theme no matter what. I'm fairly sure people would riot if Godot ever defaulted to a light theme, even at daytime only :slightly_smiling_face: 

> 2) Ok! I see why this was not done yet. I appreciate a lot that the Godot Editor is tremendously lightweight because it does not implement specific code. However, if one wants to export a project to Android the first time, Godot additionally downloads an Android package. May this be a solution for a OS-specific download?

The problem is not shipping the code in question, it's writing it in the first place. We have few contributors knowledgeable with the Windows API, and they may prefer working on other issues instead.

**Reply by theraot** [reply]
I'd say, do not detect OS, nor OS settings, leave the default as they are...

But... Have the option to configure color ramps for base color and accent color, and have Godot automatically use based on local clock time, sampled at some fixed interval (e.g. one minute, or five minutes).

Then users can set it to do whatever they want.

**Comment by Calinou** [comment]
- Superseded by https://github.com/godotengine/godot-proposals/issues/5678.
